x-pack/elastic-agent/pkg/agent/transpiler/vars.go

```diff
@@ -55,6 +59,22 @@ func (v *Vars) Replace(value string) (Node, error) {
 			}
 			set := false
 			for _, val := range vars {
+				for providerName, provider := range v.fetchContextProviders {
+					if varPrefixMatched(val.Value(), providerName) {
+						fetchProvider := provider.(composable.FetchContextProvider)
+						fval, found := fetchProvider.Fetch(val.Value())
+						if found {
+							result += value[lastIndex:r[0]] + fval
+							set = true
+							break
+						} else {
+							return NewStrVal(""), ErrNoMatch
+						}
+					}
+				}
+				if set {
+					continue
+				}
```

Comment:
I see, however no strong opinion here since I'm not super familiar with the combinations that could occur. Do you think that removing `continue` and being able to get to the `switch` block after  the `Fetch()` action would be sth that we want?

Reply:
Now that I took a closer look at this file. This is definitely in the wrong place. At its current location it would try to find a constant string in a fetch context provider, we do not want that.

Looking at it, this needs to be removed from here and moved into the:

```
func (v *Vars) Lookup(name string) (interface{}, bool) {
	return v.tree.Lookup(name)
}
```

Leaving this function untouched.

Reply:
I think I see the issue. You need to do the following:

Add the following function to `Vars`:

```
// lookupNode performs a lookup on the AST, but keeps the result as a `Node`.
//
// This is different from `Lookup` which returns the actual type, not the AST type.
func (v *Vars) lookupNode(name string) (Node, bool) {
	// check if the value can be retrieved from a FetchContextProvider
	for providerName, provider := range v.fetchContextProviders {
		if varPrefixMatched(name, providerName) {
			fetchProvider := provider.(composable.FetchContextProvider)
			fval, found := fetchProvider.Fetch(name)
			if found {
				return &StrVal{value: fval}, true
			} else {
				return &StrVal{value: ""}, false
			}
		}
	}
	// lookup in the AST tree
	return Lookup(v.tree, name)
}
```

Then change `node, ok := Lookup(v.tree, val.Value())` to `node, ok := v.lookupNode(val.Value())`.

See if that works.

Reply:
👍🏼 This fixes the issue, thank you!